### PR TITLE
feat(fs consistent snaps): integrate fs consistency code to snapshot creation flow

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
File System Consistent Snapshot

Goals:
- Issue FIFREEZE and FITHAW before and after snapshot creation.
- Filesystem to be consistent before and after snapshot creation.

Changes in this PR:
- This PR integrates the fsfreeze and unfreeze before and after snapshot creation.
- This adds an parameter to VolumeSnapshotClass to enabled/disabled fs consistency.
- fs consistency would be enabled by default on the snapshot creation.

### CODE REVIEW

File: control-plane/csi-driver/src/bin/controller/controller.rs [hunk 1]
```diff
@@ -74,6 +78,61 @@ fn parse_protocol(proto: Option<&String>) -> Result<VolumeShareProtocol, Status>
...
+        .await
+        .map_err(|error| Status::failed_precondition(error.to_string()))?;
+    match client
+        .freeze_fs(Request::new(FreezeFsRequest {
+            volume_id: volume_id.clone(),
+        }))
+        .await
+    {
+        Ok(_) => Ok(()),
+        Err(status) if status.code() == tonic::Code::Unimplemented => {
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/controller/controller.rs]: invalid argument?

File: control-plane/csi-driver/src/bin/node/fsfreeze/mod.rs [hunk 2]
```diff
@@ -49,9 +49,8 @@ fn exitcode_to_status(output: std::process::Output) -> Status {
         Some(code) if code == Errno::ENODEV as i32 || code == Errno::ENODATA as i32 => {
             Status::not_found(stderr_msg)
         }
-        Some(code) if code == Errno::EINVAL as i32 || code == Errno::EMEDIUMTYPE as i32 => {
-            Status::invalid_argument(stderr_msg)
-        }
+        Some(code) if code == Errno::EINVAL as i32 => Status::invalid_argument(stderr_msg),
+        Some(code) if code == Errno::EMEDIUMTYPE as i32 => Status::invalid_argument(stderr_msg),
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/node/fsfreeze/mod.rs]: any reason to separate these?

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/csi-driver/src/bin/node/fsfreeze/mod.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/csi-driver/src/bin/node/fsfreeze/mod.rs]: No reason, missed it. :-;

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**